src/signtool/IBM_Container.cpp

```diff
@@ -0,0 +1,1767 @@
+    printHexBytes( m_swSigQ, ECDSA521_SIG_SIZE, g_COL_WIDTH );
+
+    std::cout <<  "   m_swSigR           = ";
+    printHexBytes( m_swSigR, ECDSA521_SIG_SIZE, g_COL_WIDTH );
+    std::cout << std::endl;
+}
+
+
+// default C'tor
+IBM_Container::IBM_Container( std::string p_mode )
+   : m_containerHdr(),
+     m_prefixHdr(),
+     m_prefixData(),
+     m_softwareHdr(),
+     m_softwareSig(),
+     m_mode( p_mode )
+{
+    if ( !((m_mode == IBM_Utils::g_MODE_PRODUCTION) || 
+           (m_mode == IBM_Utils::g_MODE_DEVELOPMENT))  )
+    {
+        std::stringstream ss;
+        ss << "*** Invalid value for mode" << std::endl
+           << "--- Expecting <" << IBM_Utils::g_MODE_PRODUCTION 
+           << "> or <" << IBM_Utils::g_MODE_DEVELOPMENT << ">, got <"
+           << p_mode << ">" << std::endl;
+
+        THROW_EXCEPTION_STR(ss.str().c_str());
+    }
+
+    initializeMap();
+}
+
+
+
+// Given a filename, read its contents, parse the data and constuct the container
+IBM_Container::IBM_Container( std::string p_mode,
+                              std::string p_containerFileName )
+   : m_containerHdr(),
+     m_prefixHdr(),
+     m_prefixData(),
+     m_softwareHdr(),
+     m_softwareSig(),
+     m_mode( p_mode )
+{
+    if ( !((m_mode == IBM_Utils::g_MODE_PRODUCTION) || 
+           (m_mode == IBM_Utils::g_MODE_DEVELOPMENT))  )
+    {
+        std::stringstream ss;
+        ss << "*** Invalid value for mode" << std::endl
+           << "--- Expecting <" << IBM_Utils::g_MODE_PRODUCTION 
+           << "> or <" << IBM_Utils::g_MODE_DEVELOPMENT << ">, got <"
+           << p_mode << ">" << std::endl;
+
+        THROW_EXCEPTION_STR(ss.str().c_str());
+    }
+
+    initializeMap();
+
+    std::vector<uint8_t> buffer;
+
+    IBM_Utils* pUtils = IBM_Utils::get();
+
+    pUtils->ReadFromFile( p_containerFileName, buffer, 2048 );
+                               
+    ParseContainer( buffer );
+}
+
+
+// default D'tor
+IBM_Container::~IBM_Container()
+{
+}
+
+
+int IBM_Container::Validate()
+{
+}
+
+
+void IBM_Container::Print() const
+{
+    m_containerHdr.PrintHeader();
+    m_prefixHdr.PrintHeader();
+    m_prefixData.PrintHeader();
+    m_softwareHdr.PrintHeader();
+    m_softwareSig.PrintHeader();
+}
+
+
+bool IBM_Container::Save( const std::string p_fileName )
+{
+    // update the software key count
+    m_prefixHdr.m_swKeyCount = m_prefixData.GetSwKeyCount();
+
+    std::ofstream outfile( p_fileName.c_str(), std::ofstream::binary );
+    if (outfile.fail() )
+    {
+        std::stringstream ss;
+        ss << "!-> Failed to open file: " << p_fileName << std::endl;
+
+        THROW_EXCEPTION_STR(ss.str().c_str());
+    }
+
+    int padLen = 4096;
+
+    std::vector<uint8_t> hdrBytes;
+
+    // write the Container Header bytes
+    m_containerHdr.GetHeaderBytes( hdrBytes );
+    outfile.write( (char *) &hdrBytes[0], hdrBytes.size() );
+
+    padLen -= hdrBytes.size();
+
+    // write the Prefix Header bytes
```

Comment:
All of these kind of comments are superfluous.

If the code isn't *obviously* doing what's written in this comment, then the code needs fixing, not commenting.